cmake/FindLevelDB.cmake

```diff
@@ -0,0 +1,72 @@
+# Copyright 2017 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set(binary_dir ${FIREBASE_BINARY_DIR}/third_party/leveldb/src/leveldb)
+
+find_path(
+  LEVELDB_INCLUDE_DIR leveldb/db.h
+  HINTS
+    $ENV{LEVELDB_ROOT}/include
+    ${LEVELDB_ROOT}/include
+    ${binary_dir}/include
+  PATH_SUFFIXES leveldb
+)
+
+find_library(
+  LEVELDB_LIBRARY
+  NAMES leveldb
+  HINTS
+    $ENV{LEVELDB_ROOT}
```

Comment:
I'm mildly surprised that the paths on lines 30,31,39,40 don't end in 'lib'. (Which isn't to say that it's wrong; I'm simply unsure.)

Reply:
Good call! I hadn't actually tried this.

As a result I've:
  * handled `LEVELDB_ROOT` in `cmake/external/leveldb.cmake`
  * added the `lib` suffix here
  * removed the reference to `libmemenv.a` which we don't actually need (and didn't come with the homebrew packaging of leveldb)

I've confirmed that the build succeeds without building the leveldb external project like so:

```
brew install leveldb
cmake -H. -Bbuild -DLEVELDB_ROOT=$HOME/homebrew/Cellar/leveldb/1.20_2
make -j -Cbuild
```

And also confirmed that if you pass `LEVELDB_ROOT` pointing at a directory that doesn't exist the build will fail with output like this:

```
  Could NOT find LevelDB (missing: LEVELDB_INCLUDE_DIR LEVELDB_LIBRARY)
```